srtp_test.go

```diff
@@ -244,13 +257,13 @@ func TestRTPLifecyleInPlace(t *testing.T) {
 		}
 
 		// Copy packet, asserts that everything was done in place
-		encryptInput := make([]byte, len(decryptedRaw))
+		encryptInput := make([]byte, len(decryptedRaw), len(decryptedRaw)+10)
 		copy(encryptInput, decryptedRaw)
 
-		actualEncrypted, err := encryptContext.EncryptRTP(encryptInput, encryptInput, encryptHeader)
+		actualEncrypted, err := encryptContext.EncryptRTP(encryptInput[:0], encryptInput, encryptHeader)
```

Comment:
Why `encryptInput[:0]`? Can we do just `encryptInput`, people will look at the tests to understand how this works and might be confused by that.

This code tries to mock the [AEAD interface](https://golang.org/pkg/crypto/cipher/#AEAD) I probably messed up in a lot of places though

Reply:
AEAD will append to the buffer, not write directly to it. In fact:

```
// To reuse plaintext's storage for the encrypted output, use plaintext[:0]
// as dst. Otherwise, the remaining capacity of dst must not overlap plaintext.
```

Basically you have to decide if the API should start writing bytes at index `0` or `len(dst)`. `cipher.Stream` does the former, while `cipher.AEAD` and `hash.Hash` does the latter. I can't think of an API that both starts at index `0` and uses `append` to expand the buffer like this implementation. 

I'm actually kind of fine not changing the API in the sake of backwards compatibility. I mostly don't like passing in the `*rtp.Header` and I'm kind of indifferent to the dst different as it only matters when `len(dst) != 0`